management/queue.go

```diff
@@ -309,34 +312,54 @@ func (m *queueManager) RecentJobErrors(ctx context.Context, jobType string, wind
 		FilteredByType: true,
 		Data:           reports,
 	}, nil
+}
+
+// getJob resolves a job's information into the job that supplied the
+// information.
+func (m *queueManager) getJob(ctx context.Context, info amboy.JobInfo) (amboy.Job, error) {
+	if info.Retry.Retryable {
+		var j amboy.Job
+		var err error
+		isRetryable := amboy.WithRetryableQueue(m.queue, func(rq amboy.RetryableQueue) {
+			j, err = rq.GetAttempt(ctx, info.ID, info.Retry.CurrentAttempt)
+		})
+
+		// If the queue is retryable, return the result immediately. Otherwise,
+		// if it's not a retryable queue, then a retryable job is treated no
+		// differently from a non-retryable job (i.e. it's not retried).
+		if isRetryable {
+			return j, err
+		}
+	}
+
+	j, ok := m.queue.Get(ctx, info.ID)
+	if !ok {
+		return j, errors.New("could not find job")
```

Comment:
They should create an `errors.When` for conditional errors.

Reply:
What do you mean? There's no When method in the errors package.

Reply:
I know, I was just commenting that it would be great if there were one.